pkg/envoy/cell.go

```diff
@@ -327,12 +334,20 @@ func registerSecretSyncer(params syncerParams) error {
 		params.Config.EnvoySecretsNamespace:      func() bool { return option.Config.EnableEnvoyConfig },
 		params.Config.IngressSecretsNamespace:    func() bool { return params.Config.EnableIngressController },
 		params.Config.GatewayAPISecretsNamespace: func() bool { return params.Config.EnableGatewayAPI },
+		params.Config.PolicySecretsNamespace:     func() bool { return params.Config.EnablePolicySecretsSync },
 	} {
 		if len(namespace) > 0 && cond() {
 			namespaces[namespace] = struct{}{}
 		}
 	}
 
+	// Policy secret syncing requires notifying the included xDSServer of the configured value,
+	// so it can be correctly used by the code.
+	if params.Config.EnablePolicySecretsSync {
```

Comment:
Having the Envoy secret sync to set/propagate the config property of the namespace onto the xdsServer and SecretManager seems wrong to me.

This looks like if the Envoy component is the owner of this information - but actually it should be the other way around.

Therefore, I'd love to see that the config properties (`PolicySecretsNamespace` & `EnablePolicySecretsSync`) are defined on the SecretManager and exposed via a getter methods only. Even if the SecretManager might be deprecated and deleted in the future. (At least the namespace configuration would have to survive anyway.)

Envoy SecretSync & xdsServer should depend on the Manager and read the config where necessary. (Even though these dependencies aren't perfect either - at least this marks the Policy component (SecretManager) as the owner of this information)

wdyt?

Note: Once we would have this refactoring - I plan to do a follow up PR so that components can register a xDS secret sync the same way as it's already possible on the Operator Secret sync level. This way we can already get rid of one explicit dependency to the SecretManager (Envoy Secretsync -> SecretManager)

Reply:
I've done this refactoring in f7251f6ccbb93a4555d4d658e069bb46ba6859c2, but I'll be honest, I'm not sure about it. It feels a little cleaner to me to have all the secret sync config in the one place. PTAL and tell me what you think @mhofstetter.

Reply:
As written, in addition to moving the config, we should remove the functions `GetPolicySecretSyncNamespace` (unused anyway) & `SetPolicySecretSyncNamespace` from the `xdsServer` too. Also the config property `policySecretSyncNamespace` in the `xdsServer`.

By adding a dependency from the `xdsServer` to the `SecretManager` (the same as we have one from the `SecretSync`) we should be able to retrieve the secret namespace value when passing it on into the function `getNetworkPolicy`.

This way we have the namespace as field only in the `SecretManager`.

> It feels a little cleaner to me to have all the secret sync config in the one place

IMO it only looks like this because we have the secret sync namespaces of ingress & gw-api here as well. But IF these modules would have packages in the Cilium Agent, these config properties wouldn't live in the secretsync module either. Otherwise we would end up that the secretsync package not only keeps all the config values, but also has to configure all the actual modules (like we did here for the policy secret sync). It's the same for the Operator secret sync where we also use some form of registration mechanism.

Reply:
Okay, this is done in 83f02592cba52b9f13326cad4a18cb8e9cd2e5fb.